api/v1alpha1/gateway_types.go

```diff
@@ -182,7 +225,15 @@ type ListenerTLS struct {
 	// values.
 	//
 	// +optional
-	MinimumVersion *string `json:"minimumVersion"`
+	MinimumVersion *string `json:"minimumVersion,omitempty"`
+	// TLSTermination defines how to terminate TLS connections.
+	//
+	// If unspecified, TLS termination type "Edge" will be used.
+	//
+	// Support: Core
+	//
+	// +optional
+	TLSTermination TLSTerminationType `json:"tlsTermination,omitempty"`
```

Comment:
This gets interesting. Sometimes one can be in a situation where the termination is decided based on the SNI. For some SNIs, the proxy terminates TLS while for other it is passthrough. It has come up a few times but I'm not sure if it should be supported by our API.

Leaving a note here but I think it is okay to keep this as is.